azurelinuxagent/common/protocol/wire.py

```diff
@@ -867,6 +884,33 @@ def get_header_for_cert(self):
             "x-ms-guest-agent-public-x509-cert": cert
         }
 
+    def write_ip_on_rdma(self):
+        logger.verbose("Parsing SharedConfig XML contents for RDMA details")
+        xml_doc = parse_doc(self.shared_conf.xml_text)
+        if xml_doc is None:
+            logger.error("Could not parse SharedConfig XML document")
+            return
+        instance_elem = find(xml_doc, "Instance")
+        if not instance_elem:
+            logger.error("Could not find <Instance> in SharedConfig document")
+            return
+
+        rdma_ipv4_addr = getattrib(instance_elem, "rdmaIPv4Address")
+        if not rdma_ipv4_addr:
+            logger.error("Could not find rdmaIPv4Address attribute on Instance element of SharedConfig.xml document")
+
+        rdma_mac_addr = getattrib(instance_elem, "rdmaMacAddress")
+        if not rdma_mac_addr:
+            logger.error("Could not find rdmaMacAddress attribute on Instance element of SharedConfig.xml document")
+        rdma_mac_addr=self.format_mac_addr(rdma_mac_addr)
+        logger.info("Found RDMA details. IPv4={0} MAC={1}".format(rdma_ipv4_addr, rdma_mac_addr))
+        RDMADeviceHandler(rdma_ipv4_addr, rdma_mac_addr).start()
+
+    def format_mac_addr(self, addr):
+        """
+        Formats a MAC addr like 00155D33FF1D into "00:15:5D:33:FF:1D"
+        """
+        return ':'.join([addr[i:i+2] for i in range(0, len(addr), 2)])
```

Comment:
Hmm I am wondering why all the rdma handling would be don ein a class named "UploadError". The class name would imply that this class would handle errors when something fails to upload.